src/components/ZoomBanner.component.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react'
+import './../styles/zoomwindow.css'
+import { debounce } from 'lodash'
+import classnames from 'classnames'
+class ZoomBanner extends Component {
+  state = {
+    isShowing: false
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      scale: nextProps.scale,
+      isShowing: true
+    })
+
+    setTimeout(() => {
+      this.setState({ isShowing: false })
+    }, 1000)
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.setState({
```

Comment:
instead of calling `setState` in the `constructor`, if you need to define the initial state, you should do it in the class property `state` (line 6)